File: _정답코드/2주차/08_필요한_구명보트/Solution.java
import java.util.*;
import java.util.stream.*;

public class Solution {
    public int solution(int limit, int[] people) {
        List<Integer> peopleList = Arrays.stream(people)
            .boxed()
            .collect(Collectors.toList());
        Collections.sort(peopleList);

        int answer = 0;
        int passed = 0;

        while(passed < peopleList.size()){
            if(peopleList.get(passed) + peopleList.get(peopleList.size() - 1) <= limit) {
                passed += 1;
            }
            answer += 1;
            peopleList.remove(peopleList.size() - 1);
        }

        return answer;
    }
}

File: _정답코드/2주차/08_필요한_구명보트/solution.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int limit, vector<int> people) {
    sort(people.begin(), people.end());

	int answer = 0;
    int passed = 0;
    while (passed < people.size()){
        if(people[passed] + people[people.size()-1] <= limit){
            passed += 1;
        }
        answer += 1;
        people.pop_back();
    }
	return answer;
}

